src/main/java/mate/academy/internetshop/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,199 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.internetshop.dao.UserDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Role;
+import mate.academy.internetshop.model.User;
+import mate.academy.internetshop.util.ConnectionUtil;
+
+@Dao
+public class UserDaoJdbcImpl implements UserDao {
+    @Override
+    public Optional<User> findByLogin(String login) {
+        String selectQuery = "SELECT * FROM users "
+                + "JOIN users_roles ON users.user_id = users_roles.user_id "
+                + "JOIN roles ON users_roles.role_id = roles.role_id "
+                + "WHERE users.login = ?";
```

Comment:
while setting roles you are making additional request to DB to get roles, should we join those tables here?